src/fourier.jl

```diff
@@ -0,0 +1,755 @@
+                      two_n_ave)
+        dIm = sqrt((PrPs - cross_power.real ^ 2 + cross_power.imag ^ 2) /
+                      two_n_ave)
+        gsq = raw_coherence(cross_power, seg_power, ref_power,
+                            seg_power_noise, ref_power_noise, n_ave)
+        dphi = sqrt((1 - gsq) / (2 * gsq * n_ave))
+        dG = sqrt(PrPs / n_ave)
+    end
+
+    return dRe, dIm, dphi, dG
+end
+
+function cross_to_covariance(cross_power, ref_power, ref_power_noise, delta_nu)
+    return cross_power * sqrt(delta_nu ./ (ref_power .- ref_power_noise))
+end
+
+function _which_segment_idx_fun(;binned=false, dt=nothing)
+    if (!binned)
+        fun = generate_indices_of_segment_boundaries_unbinned
+    else
+        # Define a new function, so that we can pass the correct dt as an
+        # argument.
+        fun(args...; kwargs...) = generate_indices_of_segment_boundaries_binned(args...; dt=dt)
+    end
+    return fun
+end
+
+function get_average_ctrate(times, gti, segment_size; counts= nothing)::Float64
+    n_ph = 0
+    n_intvs = 0
+    binned = !isnothing(counts)
+    func = _which_segment_idx_fun(;binned)
+
+    for (_, _, idx0, idx1) in func(times, gti, segment_size)
+        if !(binned)
+            n_ph += idx1 - idx0
+        else
+            n_ph += sum(counts[idx0+1:idx1])
+        end
+        n_intvs += 1
+    end
+    return (n_ph / (n_intvs * segment_size))
+end
+
+@resumable function get_flux_iterable_from_segments(times, gti, segment_size; n_bin=nothing,
+                                         fluxes=nothing, errors=nothing)
+    dt = nothing
+    binned = !isnothing(fluxes)
+    if binned
+        dt = Statistics.median(diff(times[1:100]))
+    end
+    fun = _which_segment_idx_fun(;binned, dt)
+
+    for (s, e, idx0, idx1) in fun(times, gti, segment_size)
+        if idx1 - idx0 < 2
+            @yield nothing
+            continue
+        end
+        if !binned
+            event_times = times[idx0:idx1-1]
+            # astype here serves to avoid integer rounding issues in Windows,
+            # where long is a 32-bit integer.
+            cts = fit(Histogram,Float64.(event_times .- s);nbins=n_bin).weights
+        else
+            cts = Float64.(fluxes[idx0+1:idx1])
+            if !isnothing(errors)
+                cts = cts, errors[idx0+1:idx1]
+            end
+        end
+        @yield cts
+    end
+end
+
+function avg_pds_from_iterable(flux_iterable, dt; norm="frac", use_common_mean=true,
+                               silent=false)
+    local_show_progress = show_progress
+    if silent
+        local_show_progress = (a)->a
+    end
+
+    # Initialize stuff
+    cross = unnorm_cross = nothing
+    n_ave = 0
+    freq = nothing
+
+    sum_of_photons = 0
+    common_variance = nothing
+    fgt0 = nothing
+    n_bin = nothing
+
+    for flux in local_show_progress(flux_iterable)
+        if isnothing(flux) || all(flux .== 0)
```

Comment:
```suggestion
        if isnothing(flux) || all(iszero, flux)
```
has two advantages: you don't allocate the array `flux .== 0`, and also `all` is able to return early at the first non-zero element of `flux`.